content/post/43-js-ipfs-0.32.md

```diff
@@ -0,0 +1,52 @@
+---
+date: 2018-09-11
+url: 43-js-ipfs-0-32
+title: js-ipfs 0.32.0 released
+author: Alan Shaw
+---
+
+A pinch of IPNS, some chunking of files and you'll have a delicious IPFS stew for supper
+
+# 🔦 Highlights
+
+## 📛 IPNS locally
+
+IPNS has it's foot in the door, and it's coming in! In this release IPNS works for your local node. It means you can now [publish](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/NAME.md#namepublish) records to your local repo and [read](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/NAME.md#nameresolve) them back. There's still some way to go but you know the saying right, "from small beginnings comes great IPNS"?
+
+## 🕵️‍♀️ Partial `ipfs.resolve`
+
+In preparation for IPNS landing, we've rolled out a partial implementation of the [`resolve`](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/MISCELLANEOUS.md#resolve) command. The idea of the resolve command is to figure out the underlying hash of the content referred to by an IPNS name or an IPFS path.
+
+The new resolve command can resolve paths like `/ipfs/QmRootHash/path/to/file` to `/ipfs/QmFileHash` and will soon be able to deal with IPNS names in an upcoming release.
+
+## 📦 libp2p bundle function
+
+We made it even easier to create your libp2p bundle! If you're looking to create a completely custom libp2p bundle then you can now pass a [function in place of your libp2p config](https://github.com/ipfs/js-ipfs#optionslibp2p) which should return your libp2p bundle. It's passed useful information like the IPFS node peer ID so you should be able to build your bundle exactly the way you want! You're welcome 😁
+
+## 🥒 Support for chunking algorithm
+
+Slice and dice your DAG nodes the way _you_ want™️. When adding data to your IPFS node you can now specify the size of the chunks it creates from your data or even use [`rabin` fingerprinting](https://en.wikipedia.org/wiki/Rabin_fingerprint) to create some fancy variable length chunks for better deduping.
+
+[Rabin](https://www.npmjs.com/package/rabin) is an [optional dependency](https://docs.npmjs.com/files/package.json#optionaldependencies), which means that to use it you'll have to install it in your project first with `npm install rabin`.
```

Comment:
This sounds more like instructions for peer dependencies in npm > 3 rather than optional dependencies

Reply:
👍 thanks, will fix